rust/osm2lanes/src/transform/tags_to_lanes/mod.rs

```diff
@@ -566,58 +568,75 @@ pub fn tags_to_lanes(
 }
 
 fn driving_lane_directions(tags: &Tags, _locale: &Locale, oneway: Oneway) -> (usize, usize) {
-    let both_ways = if let Some(n) = tags
-        .get("lanes:both_ways")
-        .and_then(|num| num.parse::<usize>().ok())
-    {
-        n
-    } else {
-        0
-    };
-    let num_driving_fwd = if let Some(n) = tags
+    let tagged_lanes = tags.get("lanes").and_then(|num| num.parse::<usize>().ok());
+    let tagged_forward = tags
         .get("lanes:forward")
-        .and_then(|num| num.parse::<usize>().ok())
-    {
-        n
-    } else if let Some(n) = tags.get("lanes").and_then(|num| num.parse::<usize>().ok()) {
-        let half = if oneway.into() {
-            n
-        } else {
-            // usize division rounded up
-            (n + 1) / 2
-        };
-        half - both_ways
-    } else if tags.is("lanes:bus", "2") {
-        2
-    } else {
-        1
-    };
-    let num_driving_back = if let Some(n) = tags
+        .and_then(|num| num.parse::<usize>().ok());
+    let tagged_backward = tags
         .get("lanes:backward")
-        .and_then(|num| num.parse::<usize>().ok())
-    {
-        n
-    } else if let Some(n) = tags.get("lanes").and_then(|num| num.parse::<usize>().ok()) {
-        let base = n - num_driving_fwd;
-        let half = if oneway.into() {
-            base
-        } else {
-            // lanes=1 but not oneway... what is this supposed to mean?
-            base.max(1)
-        };
-        half - both_ways
-    } else if tags.is("lanes:bus", "2") {
-        if oneway.into() {
-            1
-        } else {
-            2
+        .and_then(|num| num.parse::<usize>().ok());
+    let tagged_both_ways = tags
+        .get("lanes:both_ways")
+        .and_then(|num| num.parse::<usize>().ok());
+    let num_both_ways = tagged_both_ways.unwrap_or(0);
+    let tagged_bus_lanes = tags
+        .get("lanes:bus")
```

Comment:
comment for myself: I think we look at this tag again in busway processing, need to see if this is a code smell or something inherent to the problem.

Reply:
It makes sense to me, to look at it is both places if this function is trying to determine the number of lanes definitively. The `lanes:bus` tag is only being looked at because the way is under tagged, and we want to guess that there are some unrestricted lanes as well. Which would suggest that we should potentially look at `busway` tags here too.